aptos-move/block-executor/src/proptest_types/types.rs

```diff
@@ -1063,224 +926,96 @@ pub(crate) enum GroupSizeOrMetadata {
-                    size.get(),
-                    resources.get(key).unwrap().bytes().map_or(0, |b| b.len()) as u64
-                );
-            }
-        }
-
-        assert_ok!(self.materialized_delta_writes.set(aggregator_v1_writes));
-        // TODO[agg_v2](tests): Set the patched resource write set and events. But that requires the function
-        // to take &mut self as input
-        Ok(())
-    }
-
-    fn set_txn_output_for_non_dynamic_change_set(&self) {
-        // TODO[agg_v2](tests): anything to be added here for tests?
-    }
-
-    fn fee_statement(&self) -> FeeStatement {
-        // First argument is supposed to be total (not important for the test though).
-        // Next two arguments are different kinds of execution gas that are counted
-        // towards the block limit. We split the total into two pieces for these arguments.
-        // TODO: add variety to generating fee statement based on total gas.
-        FeeStatement::new(
-            self.total_gas,
-            self.total_gas / 2,
-            (self.total_gas + 1) / 2,
-            0,
-            0,
-        )
-    }
+// ID is just the unique index as u128.
+pub(crate) fn serialize_from_delayed_field_u128(value_or_id: u128, version: u32) -> Bytes {
+    let tuple = (value_or_id, version);
+    serialize_delayed_field_tuple(&tuple)
+}
 
-    fn output_approx_size(&self) -> u64 {
-        // TODO add block output limit testing
-        0
-    }
+pub(crate) fn serialize_from_delayed_field_id(
+    delayed_field_id: DelayedFieldID,
+    version: u32,
+) -> Bytes {
+    let tuple = (delayed_field_id.extract_unique_index() as u128, version);
+    serialize_delayed_field_tuple(&tuple)
+}
 
-    fn get_write_summary(
-        &self,
-    ) -> HashSet<
-        crate::types::InputOutputKey<
-            <Self::Txn as Transaction>::Key,
-            <Self::Txn as Transaction>::Tag,
-        >,
-    > {
-        HashSet::new()
-    }
+/// Serializes a (u128, u32) tuple into Bytes using BCS serialization
+fn serialize_delayed_field_tuple(value: &(u128, u32)) -> Bytes {
+    bcs::to_bytes(value)
+        .expect("Failed to serialize (u128, u32) tuple")
+        .into()
 }
 
-#[derive(Clone, Debug)]
-pub(crate) struct MockEvent {
-    event_data: Vec<u8>,
+/// The width of the delayed field is not used in the tests, and fixed as 8 for
+/// all delayed field constructions. However, only the real ID is actually
+/// serialized and deserialized (together with the version).
+pub(crate) fn deserialize_to_delayed_field_u128(bytes: &[u8]) -> Result<(u128, u32), bcs::Error> {
+    bcs::from_bytes::<(u128, u32)>(bytes)
 }
 
-impl TransactionEvent for MockEvent {
-    fn get_event_data(&self) -> &[u8] {
-        &self.event_data
-    }
+pub(crate) fn deserialize_to_delayed_field_id(
+    bytes: &[u8],
+) -> Result<(DelayedFieldID, u32), bcs::Error> {
+    let (id, version) = bcs::from_bytes::<(u128, u32)>(bytes)?;
+    Ok((DelayedFieldID::from((id as u32, 8)), version))
```

Comment:
What is `8`? Width?

Reply:
yeah, the width of ID. it was kinda confusing but it works like this